spec/acceptance/candlepin_spec.rb

```diff
@@ -163,4 +163,49 @@ class { 'certs::candlepin':
     its(:stdout) { should match(/^Owner: CN=localhost, OU=SomeOrgUnit, O=Katello, ST=North Carolina, C=US$/) }
     its(:stdout) { should match(/^Issuer: CN=#{host_inventory['fqdn']}, OU=SomeOrgUnit, O=Katello, L=Raleigh, ST=North Carolina, C=US$/) }
   end
+
+  context 'updates java-client certificate in truststore if it changes' do
+    let(:pp) do
+      <<-EOS
+      user { 'tomcat':
+        ensure => present,
+      }
+
+      ['/usr/share/tomcat/conf', '/etc/candlepin/certs'].each |$dir| {
+        exec { "mkdir -p ${dir}":
+          creates => $dir,
+          path    => ['/bin', '/usr/bin'],
+        }
+      }
+
+      package { 'java-1.8.0-openjdk-headless':
+        ensure => installed,
+      }
+
+      include certs::candlepin
+      EOS
+    end
+
+    before(:all) do
+      pp_setup = <<-EOS
+        exec { "rm -rf /root/ssl-build/localhost":
+          path => "/bin:/usr/bin",
+        }
+      EOS
+
+      apply_manifest(pp_setup, catch_failures: true)
+    end
+
+    it_behaves_like 'a idempotent resource'
+
+    it "checks that the fingerprint matches" do
+      keytool = <<-EOS
+        exec { "keytool -list -keystore /etc/candlepin/certs/truststore -storepass $(cat #{truststore_password_file}) | grep $(openssl x509 -noout -fingerprint -in /etc/pki/katello/certs/java-client.crt | cut -d '=' -f 2)":
+          path => "/bin:/usr/bin",
+        }
+      EOS
+
+      apply_manifest(keytool, catch_failures: true)
```

Comment:
Is the [command](https://serverspec.org/resource_types.html#command) resource perhaps better?

Reply:
It did not seem to handle pipes

Reply:
Then the more correct thing is
```ruby
it "checks that the fingerprint matches" do
  fingerprint_output = on default, 'openssl x509 -noout -fingerprint -in /etc/pki/katello/certs/java-client.crt'
  fingerprint = fingerprint_output.split('=').last
  expect(on default, "keytool -list -keystore /etc/candlepin/certs/truststore -storepass file:#{truststore_password_file}").to include(fingerprint)
end
```